### PR TITLE
feat: test execution crd

### LINKED ISSUE #4017: Trigger new test execution from CR
**Is your feature request related to a problem? Please describe.**

As a testkube user 
In order to trigger a test execution from a gitops K8S controller (as a workaround for #4007 and #3712)
I need testkube to offer a way to trigger a test from a k8s resource

**Describe the solution you'd like**

A new TestExecution CRD that can trigger a testexecution + provide status on its execution

**Describe alternatives you've considered**

Use the tk rest api or tk cli. 

* With fluxcd, the generic webhook notification https://fluxcd.io/flux/components/notification/provider/#generic-webhook don't seem to allow for the required http request transformation to match the tk api. A testkube [flux provider](https://fluxcd.io/flux/components/notification/provider/) could be developped in go
* With kapp controller, would require instead running a Job with the tk cli being invoked. 

**Additional context**

https://discord.com/channels/884464549347074049/1113808657843507200/1116703138469138503

### PR DESCRIPTION
**Pull request by vsukhin**
## Pull request description 

Fix #4017

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by nicufk — APPROVED** [review]
LGTM, needs a review from @jfermi for the documentation

File: docs/docs/articles/test-executions.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Test and Test Suite Execution CRDs
+
+Testkube allows you to automatically run tests and test suites by creating or updating Test or Test Suite Execution CRDs.
+
+## What is a Testkube Execution CRDs?
```
Comment (jfermi) [inline docs/docs/articles/test-executions.md:5]:
```suggestion
## What are Testkube Execution CRDs?
```

File: docs/docs/articles/test-executions.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    variables:
+      VAR_TEST:
+        name: VAR_TEST
+        value: "ANY"
+        type: basic
+```
+
+## Architecture
+
+Testkube uses Kubernets Operator to reconcile Test and Test Suite Execution CRDs state and run corresponding test and test suite when resource generation is changed.
```
Comment (jfermi) [inline docs/docs/articles/test-executions.md:69]:
```suggestion
Testkube uses Kubernetes Operator to reconcile Test and Test Suite Execution CRDs state and run the corresponding test and test suite when resource generation is changed.
```

File: docs/docs/articles/test-executions.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+or 
+
+```yaml
+testSuite:
+  name: Testkube test suite name
+```
+
+### Execution Request
+
+Execution Request allows to define execution parameters for each specific resource.
```
Comment (jfermi) [inline docs/docs/articles/test-executions.md:27]:
```suggestion
An Execution Request defines execution parameters for each specific resource.
```

**Review by jfermi — COMMENTED** [review]
A few editing suggestions.

**Pull request merged**